source/tutorials/learning-journey/shell-dot-nix.md

```diff
@@ -0,0 +1,163 @@
+So, if you properly describe the shell environment you want in a `shell.nix` file, you can enter it with just the `nix-shell` command without any further arguments.
+No more specifying packages on the command line.
+Here's what a basic `shell.nix` looks like that installs Python 3.10 as before:
+```nix
+let
+  pkgs = import <nixpkgs> {};
+in
+  pkgs.mkShell {
+    packages = [
+      pkgs.python3
+    ];
+  }
+```
+where `mkShell` is a function that when called produces a shell environment.
+
+If you save this into a file called `shell.nix` and call `nix-shell` in the directory containing this `shell.nix` file, you'll enter a shell with Python 3.10 installed.
+
+## Adding packages
+Additional executable packages are added to the shell by adding them to the `packages` attribute.
+For example, let's say we wanted to add `curl` to our shell environment.
+The new `shell.nix` would look like this:
+```nix
+let
+  pkgs = import <nixpkgs> {};
+in
+  pkgs.mkShell {
+    packages = [
+      pkgs.python3
+      pkgs.curl  # new package
+    ];
+  }
+```
+
+:::{note}
+`nix-shell` was originally conceived as a way to construct a shell environment containing the tools needed to *develop software*; only later was it widely used as a general way to construct temporary environments for other purposes. Also note that `mkShell` is a [wrapper around `mkDerivation`](https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-mkShell) so strictly speaking you can provide any attributes to `mkShell` that you could to `mkDerivation` such as `buildInputs`. However, the `packages` attribute provided to `mkShell` is an alias for `buildInputs`, so you shouldn't need to provide both `packages` and `buildInputs`.
+:::
+
+## Environment variables
+It's common to want to automatically export certain environment variables when you enter a shell environment.
+For example, you could have a database that depends on an environment variable to set the default authentication credentials during development.
+
+Setting an environment variable in via `shell.nix` is trivial.
+Any attribute in the `mkShell` function call that `mkShell` doesn't recognize as a reserved attribute name will be set to an environment variable in the shell environment.
+The attributes that are reserved are listed in the [Nixpkgs manual][mkshell_attrs] and include `packages`, `name`, and several others.
+
+[mkshell_attrs]: https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-mkShell-attributes
+
+Let's say you wanted to set the database user (`DB_USER`) and password (`DB_PASSWORD`) via environment variables in your `shell.nix` file.
+This is how that would look:
+```nix
+let
+  pkgs = import <nixpkgs> {};
+in
+  pkgs.mkShell {
+    packages = [
+      pkgs.python310
+      pkgs.curl
+    ];
+
+    env = {
+      # Database credentials
+      DB_USER = "db_user";
+      DB_PASSWORD = "super secret don't look";
+    };
+  }
+```
+
+
```

Comment:
Discussed online: mention this as a warning rather than guidance. Mention that you can use `shellHook` to still set these variables if you _need_ to. Then illustrate the Bash syntax for exporting environment variables to do this as a one line example. Another example is setting secrets you get from another command.